### DISCUSSION TITLE
AG CPU high usage in Logic pro

### DISCUSSION THREAD

**Discussion by abab6111** [apohl79/audiogridder#1385] [Help]
My device: Mac mini M2, 24GB RAM, Ventura 13.5

Logic pro Version: 11.0.0

Connect to Server: Windows 10, i5-12400, 32GB RAM

When I load AG plugin(No any instrument) in Logic pro, CPU usage will be High.
This image shows obviously CPU usage is high after loading AG plugin.
<img width="294" alt="CleanShot 2024-08-14 at 23 32 09@2x" src="https://github.com/user-attachments/assets/d23f4c2c-66df-4331-bf69-cc77b784d698">

After I remove AG plugin, CPU usage fell immediately.
<img width="275" alt="CleanShot 2024-08-14 at 23 32 44@2x" src="https://github.com/user-attachments/assets/12ce0144-338c-4ea2-a504-7a81530837db">

In Activity Monitor, Most of the CPU occupied by AUhostingService(Logic pro) and AudiogridderPluginTray 
<img width="820" alt="CleanShot 2024-08-14 at 23 28 00@2x" src="https://github.com/user-attachments/assets/41cdf9f8-0828-4fb1-b390-3e6b206e48ad">

No matter how I adjust buffer size and block, It still can't solve it.

**Comment by Kramer2025** [comment]
Try fixed size buffers. This helped me from spikes to 60% down to 0% idle

**Reply by abab6111** [reply]
Could you share how to set it?
No matter how I adjust the buffer size, CPU usage is still very high.

**Reply by Kramer2025** [reply]
I am using a built in feature in my DAW for fixed size buffers in FL studio plug-in wrapper. This is the only thing that worked for me instantly. I don't think logic has anything like that.

You can increase buffers to 4096 from AG  VST/AU and try to experiment with "fixed size outbound buffers" button. 

Another thing that helped me in past was using Sandbox mode. You can find that in your server settings. Any option might work but try then both and see for yourself. 

Another thing that might be a problem, is the ethernet settings on your windows machine. There are no definite answers for this but experimenting. 

Chat GTP can be helpful on where to look if you ask it that you want to drop latency. 
I remember asking it about AG specifically and it gave me some answers but I am not sure how valid they were.
